src/IO/S3/copyS3File.cpp

```diff
@@ -250,6 +250,62 @@ namespace
             multipart_upload_aborted = true;
         }
 
+        void scheduleAbortMultipartUpload()
+        {
+            bool expected = false;
+            if (!multipart_upload_aborted.compare_exchange_strong(expected, true))
+                return;
+
+            if (schedule)
+            {
+                {
+                    std::lock_guard lock(bg_tasks_mutex);
+                    ++num_added_bg_tasks;
+                }
+
+                auto task_finish_notify = [this]()
+                {
+                    std::lock_guard lock(bg_tasks_mutex);
+                    ++num_finished_bg_tasks;
+
+                    bg_tasks_condvar.notify_one();
```

Comment:
may be it is a time to use here `src/Common/ThreadPoolTaskTracker.h` instead manual managment

Reply:
Look, it really makes things more simple here.
https://github.com/ClickHouse/ClickHouse/pull/86764